sources/api/shibaken/src/warmpool/autoscaling_warm_pool.rs

```diff
@@ -0,0 +1,132 @@
+/// This module contains utilities for waiting until the instance on warm pool to be
+/// marked as InService.
+use crate::warmpool::error::{self, Result};
+use argh::FromArgs;
+use imdsclient::ImdsClient;
+use log::{info, warn};
+use serde::Deserialize;
+use snafu::{OptionExt, ResultExt};
+use std::fs;
+use std::path::Path;
+use tokio::time::{sleep, Duration};
+
+// Path to config file containing host's autoscaling.should-wait setting
+const CONFIG_PATH: &str = "/etc/warm-pool-wait.toml";
+const FETCH_LIFECYCLE_INTERVAL_SECS: Duration = Duration::from_secs(5);
+
+#[derive(FromArgs, Debug)]
+#[argh(subcommand, name = "warm-pool-wait")]
+/// Fetch to IMDS and wait until instance is marked as InService.
```

Comment:
Nit:

I would say that `Fetch to IMDS` is irrelevant here:

```suggestion
Wait until the instance reaches the `InService` state
```